icu4j/main/tests/core/src/com/ibm/icu/dev/test/message2/Mf2FeaturesTest.java

```diff
@@ -0,0 +1,461 @@
+
+    @Before
+    public void prepareDefaultLocale() {
+        Locale.setDefault(Locale.US);
+    }
+
+    @Test
+    public void testEmptyMessage() {
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern("{}")
+                .arguments(Args.NONE)
+                .expected("")
+                .build());
+    }
+
+    @Test
+    public void testPlainText() {
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern("{Hello World!}")
+                .arguments(Args.NONE)
+                .expected("Hello World!")
+                .build());
+    }
+
+    @Test
+    public void testPlaceholders() {
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern("{Hello, {$userName}!}")
+                .arguments(Args.of("userName", "John"))
+                .expected("Hello, John!")
+                .build());
+    }
+
+    @Test
+    public void testArgumentMissing() {
+        // Test to check what happens if an argument name from the placeholder is not found
+        // We do what the old ICU4J MessageFormat does.
+        String message = "{Hello {$name}, today is {$today :datetime skeleton=yMMMMdEEEE}.}";
+
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern(message)
+                .arguments(Args.of("name", "John", "today", TEST_DATE))
+                .expected("Hello John, today is Wednesday, November 23, 2022.")
+                .build());
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern(message)
+                .arguments(Args.of("name", "John"))
+                .expected("Hello John, today is {$today}.")
+                .build());
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern(message)
+                .arguments(Args.of("today", TEST_DATE))
+                .expected("Hello {$name}, today is Wednesday, November 23, 2022.")
+                .build());
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern(message)
+                .arguments(Args.NONE)
+                .expected("Hello {$name}, today is {$today}.")
+                .build());
+    }
+
+    @Test
+    public void testDefaultLocale() {
+        String message = "{Date: {$date :datetime skeleton=yMMMMdEEEE}.}";
+        String expectedEn = "Date: Wednesday, November 23, 2022.";
+        String expectedRo = "Date: miercuri, 23 noiembrie 2022.";
+        Map<String, Object> arguments = Args.of("date", TEST_DATE);
+
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern(message)
+                .arguments(arguments)
+                .expected(expectedEn)
+                .build());
+        TestUtils.runTestCase(new TestCase.Builder()
+                .pattern(message)
+                .arguments(arguments)
+                .locale("ro")
+                .expected(expectedRo)
+                .build());
+
+        Locale.setDefault(Locale.forLanguageTag("ro"));
```

Comment:
ditto restore the original

Reply:
Done. Removed the `@Before` and doing set + restore in one method only.